add-ons/updating/tb78/convert-legacy.md

```diff
@@ -0,0 +1,146 @@
+* **locale** localization for WebExtensions is handled using the [`i18n` API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/i18n), which uses a single `messages.json` file to store translations.
+* **style** use an experiment to monitor open windows, and inject the style through that experiment
+* **overlay** [see section 'Replacing Overlays' below](#replacing-overlays).
+
+There are no direct equivalents to manifest flags, so add-ons now need to provide their own mechanisms to switch code or resources depending on the runtime environment. Relevant information is accessible through the [`runtime` API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/).
+
+
+## Replacing Overlays
+
+If you used XUL overlays, they are now finally gone for good and you must find an alternative. While it is technically possible to write an XUL overlay loader experiment, it is a better idea to re-evaluate the purpose of the overlay:
+* For overlays extending the user interface in a way that can be replaced using calls to built-in APIs: do those calls in a background script (Example: adding context menu items)
+* For overlays extending the user interface in a generalizeable way: add an experiment with an API for the generic customization (Example: adding toolbar buttons to a toolbar without an existing API; use existing toolbar APIs as reference)
+* For overlays loading a script without user interface relationship: move the script's content to a background script (Example: startup script in a main window overlay)
+
+If a part of an overlay cannot be replaced with one of the suggestions above, it may be reasonable to replace the overlay with a specific function in an experiment.
+
+
+## Replacing XUL windows and dialogs
+
+While it would be possible to attempt to re-use existing XUL code in an experiment, it is probably a better idea to use the more future-proof [`windows` MailExtension API](https://thunderbird-webextensions.readthedocs.io/en/latest/windows.html) to create a window displaying an HTML dialog. From these dialogs, all WebExtension and MailExtension APIs can be accessed in the same way as from a background script.
+
+Note that current builds of Thunderbird will, by default, display an ugly URL bar on top of all windows created through the `windows` API. Specifying the type `panel` or `detached_panel` may hide the URL bar in some future configurations. Either way, the HTML of the dialog will most likely be re-useable with any future mechanism to open windows.
```

Comment:
> Specifying the type `panel` or `detached_panel` may hide the URL bar in some future configurations

I just stumbled accross that today. Where do you got those info from? It would be so great if that URL bar would be gone...

Reply:
It is a documented type that is currently disabled, but there is no reason why it couldn't get enabled again. But there have been some recent developments (https://bugzilla.mozilla.org/show_bug.cgi?id=1644961) – so this information is already obsolete. :)